app/bin/run-ecs-task/run-task.sh

```diff
@@ -0,0 +1,138 @@
+    echo "ex: ./run-task.sh -a bob.smith -e test -t db-migrate-up"
+    echo "See the README.md for further details"
+    exit 1
+}
+
+# This parses CLI arguments mapping the letter params to variables
+# Note that if you want to add more params, you need to add the letter
+# to the string next to getopts below
+while getopts ":a:e:t:s:g:" opt; do
+    case $opt in
+        a)
+            AUTHOR="$OPTARG"
+            ;;
+        e)
+            ENV_NAME="$OPTARG"
+            ;;
+        t)
+            TASK_NAME="$OPTARG"
+            ;;
+        s)
+            SUBNETS="$OPTARG"
+            ;;
+        g)
+            SECURITY_GROUPS="$OPTARG"
+            ;;
+        \?)
+            echo "Invalid option -$OPTARG"
+            exit 1
+            ;;
+    esac
+
+    # Error if an argument has no parameter
+    case $OPTARG in
+        -*) 
+            echo "Option $opt needs a valid argument"
+            exit 1
+            ;;
+    esac
+done
+
+# If the required environment variables are missing, print the usage info and error.
+if [ -z "$ENV_NAME" ] || [ -z "$TASK_NAME" ] || [ -z "$AUTHOR" ]; then
+  usage
+fi
+
+# If no security groups were specified, fetch them from AWS. The response is formatted as
+# {
+#   "SecurityGroups": [
+#       {"GroupName": "csv-handler", GroupId: "sg-123456789", ...}  
+# ]}
+if [ -z "$SECURITY_GROUPS" ]; then
+    # The jq command grabs all returned security group IDs and formats it into a JSON array
+    SECURITY_GROUPS=$(aws ec2 describe-security-groups --group-names csv-handler | jq -r --compact-output "[.SecurityGroups[].GroupId]")
+fi
+
+# If no subnets specified, fetch them from AWS. The response is formatted as
+# {
+#   "Subnets": [
+#   {"SubnetId": "subnet-012345abc6789", ...},
+#   {"SubnetId": "subnet-abcdef123ghij", ...}
+# ]}
+if [ -z "$SUBNETS" ]; then
+    # TODO - potentially add a VPC to filter too
+    # The jq command grabs all returned subnet IDs and formats it into a JSON array
+    SUBNETS=$(aws ec2 describe-subnets | jq -r --compact-output "[.Subnets[].SubnetId]")
+fi
+
+# Using the network config template, populate it with
+# either the supplied security group and subnet values
+# or the ones we fetched from AWS.
+NETWORK_CONFIG=$(jq \
+    --argjson SECURITY_GROUPS "$SECURITY_GROUPS" \
+    --argjson SUBNETS "$SUBNETS" \
+    '.awsvpcConfiguration.securityGroups=$SECURITY_GROUPS |
+     .awsvpcConfiguration.subnets=$SUBNETS' \
+    $DIR/network_config.json.tpl)
+
+# Use the container override file as-is
+OVERRIDES=$(jq . $DIR/container_overrides.json.tpl)
+
+# Build the ECS run-task command
+AWS_ARGS=("--region=us-east-1"
+    ecs run-task
+    "--cluster=$ENV_NAME"
+    "--started-by=$AUTHOR"
+    "--task-definition=$TASK_NAME"
+    "--launch-type=FARGATE"
+    "--platform-version=1.4.0"
+    --network-configuration "$NETWORK_CONFIG"
+    --overrides "$OVERRIDES"
+    )
+
+# Print out the arguments
+printf " ... %s" "${AWS_ARGS[@]}"
```

Comment:
Could this contain secrets at all? Such as from the environment variables in the overrides json?

Reply:
Technically yes, although would that be an issue for someone running locally if they just manually entered them?

I don't think you can even override parameter store env variables due to how they get loaded into the container (at runtime?) - or at least I remember running into that issue before.